src/cascadia/TerminalSettingsEditor/ViewModelHelpers.h

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+#include "../inc/cppwinrt_utils.h"
+
+template<typename T>
+struct ViewModelHelper
+{
+public:
+    winrt::event_token PropertyChanged(::winrt::Windows::UI::Xaml::Data::PropertyChangedEventHandler const& handler)
+    {
+        return _propertyChangedHandlers.add(handler);
+    }
+
+    void PropertyChanged(winrt::event_token const& token)
+    {
+        _propertyChangedHandlers.remove(token);
+    }
+
+protected:
+    void _NotifyChangeCore(const std::wstring_view name)
+    {
+        _propertyChangedHandlers(*static_cast<T*>(this), ::winrt::Windows::UI::Xaml::Data::PropertyChangedEventArgs{ name });
+    }
+
+    // template recursion base case: single dispatch
+    void _NotifyChanges(const std::wstring_view name) { _NotifyChangeCore(name); }
+
+    template<typename... Args>
+    void _NotifyChanges(const std::wstring_view name, Args&&... more)
+    {
+        _NotifyChangeCore(name);
+        _NotifyChanges(std::forward<Args>(more)...);
+    }
+
+private:
+    winrt::event<::winrt::Windows::UI::Xaml::Data::PropertyChangedEventHandler> _propertyChangedHandlers;
+};
+
+#define _BASE_OBSERVABLE_PROJECTED_SETTING(target, name)  \
+public:                                                   \
+    auto name() const noexcept { return target.name(); }; \
+    template<typename T>                                  \
+    void name(const T& value)                             \
+    {                                                     \
+        if (name() != value)                              \
+        {                                                 \
+            target.name(value);                           \
+            _NotifyChanges(L"Has" #name, L#name);         \
```

Comment:
this notifies for "HasX" and "X" simultaneously

Reply:
omg, a macro for a template. You've gone wild with power hahaha